c7n/resources/mq.py

```diff
@@ -83,10 +85,58 @@ def get_dimensions(self, resource):
                  'Value': "{}-1".format(resource['BrokerName'])}]
 
 
+@MessageBroker.filter_registry.register('vpc')
+class VpcFilter(VpcFilter):
+    """Filter a resource by its VPC id. Choose to select resources that are in that VPC or select
+    resources that are not in that VPC using the 'op' field. Security group ID is used to get
+    the VPC ID of a message broker, since DescribeMessageBrokers does not return a VPC ID.
+    Since message brokers are exclusively launched in a single VPC, and their security groups
+    all belong to this VPC, only the first security group id is needed to check the VPC
+    ID of the message broker. Continue is used in the function if a client error occurs
+    when trying to get a broker's vpc -- this will skip this broker, but still attempt to grab
+    the vpc ids of the other brokers in the list.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mq-vpc-filters
+                resource: message-broker
+                filters: [{'type': 'vpc', 'key': 'VpcId', 'value': 'vpc-xxxxxxxxxxx', 'op': 'eq'}]
```

Comment:
please format this in yaml with whitespace rather than with brackets for readability